src/families/tron/bridge/js.js

```diff
@@ -259,20 +240,90 @@ const currencyBridge: CurrencyBridge = {
   scanAccountsOnDevice
 };
 
-const createTransaction = a => {
-  throw new CurrencyNotSupported("tron currency not supported", {
-    currencyName: a.currency.name
-  });
-};
+const createTransaction = () => ({
+  family: "tron",
+  amount: BigNumber(0),
+  mode: "send",
+  duration: 3,
+  recipient: "",
+  networkInfo: null
+});
 
 const updateTransaction = (t, patch) => ({ ...t, ...patch });
 
-const getTransactionStatus = a =>
-  Promise.reject(
-    new CurrencyNotSupported("tron currency not supported", {
+const getEstimatedFees = async t => {
+  // see : https://developers.tron.network/docs/bandwith#section-bandwidth-points-consumption
+  // 1. cost around 200 Bandwidth, if not enough check Free Bandwidth
+  // 2. If not enough, will cost some TRX
+  // 3. normal transfert cost around 0.002 TRX
+  // Special case: If activated an account, cost around 0.1 TRX
+  let fees = BigNumber(0);
+
+  // Calculate Bandwidth :
+  if (t.networkInfo) {
+    const {
+      freeNetUsed = 0,
+      freeNetLimit = 0,
+      NetUsed = 0,
+      NetLimit = 0
+    } = t.networkInfo;
+    const bandwidth = freeNetLimit - freeNetUsed + NetLimit - NetUsed;
+
+    if (bandwidth < AVERAGE_BANDWIDTH_COST) {
+      fees = fees.plus(BigNumber(2000));
+      // Cost is around 0.002 TRX
+    }
+  }
+
+  // Active the account if there's no data, that require fees
+  if (t.recipient) {
+    const recipientAccount = await fetchTronAccount(t.recipient);
+    if (recipientAccount.length === 0) {
+      fees = fees.plus(BigNumber(100000));
+      // Cost is around 0.1 TRX
+    }
+  }
+
+  return fees;
+};
+
+const getTransactionStatus = async (a, t) => {
+  const errors = {};
+  const warnings = {};
```

Comment:
not really typesafe: it infers to `any` instead of:
```
  errors: { [string]: Error },
  // array of field based warning messages for a transaction
  warnings: { [string]: Error },
```